internal/services/azapi_resource.go

```diff
@@ -853,6 +871,18 @@ func (r *AzapiResource) ImportState(ctx context.Context, request resource.Import
 	response.Diagnostics.Append(response.State.Set(ctx, state)...)
 }
 
+func (r *AzapiResource) isPreflightSupported(isNewResource bool, parentID string) bool {
+	// currently, parentID must be one of resource group, subscription, management group or tenant('/'). extension resources and nested resources are not supported by Preflight.
+	resourceType := utils.GetResourceType(parentID)
+
+	isSupportedParentID := strings.EqualFold(arm.ResourceGroupResourceType.String(), resourceType) ||
+		strings.EqualFold(arm.SubscriptionResourceType.String(), resourceType) ||
+		strings.EqualFold(arm.TenantResourceType.String(), resourceType) ||
+		strings.EqualFold("Microsoft.Management/managementGroups", resourceType)
+
+	return r.ProviderData.Features.EnablePreflight && isNewResource && isSupportedParentID
```

Comment:
I think `r.ProviderData.Features.EnablePreflight && isNewResource` should be moved out from this function, because this function is about "whether preflight is supported on a give resource", once this is addressed, this PR should be good to go. Thanks!